### ISSUE TITLE
Manage event(s): Update event

### ISSUE THREAD

**Issue by lauren-mills** [microsoft/botframework-skills#308]
## User Story
**As an** employee at an Enterprise
**I want** VA to update an event for me
**so that** I can add a participant/change the date/time and location and feel secure about my upcoming meetings.

## Description
Dependent on #304 

## Conversation Design
[Design](https://www.figma.com/file/xsSfVmzSe5CyFlcLovplS9/Calendar-Skill?node-id=215%3A15)
[Utterances and LG responses](https://microsoft.sharepoint-df.com/:x:/t/BotFrameworkSolutions/Eer8FvZT2DhPmYMBfRVa7X4Bf9R0w6Bm9JJn5y8sd50g4A?e=VmbzTe)

## Language Understanding
| Intent | Sample Utterances | Prebuilt Entities | Custom Entities | Patterns |
| ------ | ----------------------| ----------------- | ----------------- | --------- |
| UpdateEvent | [Excel](https://microsoft.sharepoint-df.com/:x:/t/BotFrameworkSolutions/Eer8FvZT2DhPmYMBfRVa7X4Bf9R0w6Bm9JJn5y8sd50g4A?e=VmbzTe) | datetimev2, personName, emailAddress | title, location | - |
| UpdateEventTitle | | | | | |
| UpdateEventDateTime | | | | | |
| UpdateEventAddAttendee | | | | | |
| UpdateEventRemoveAttendee | | | | | |
| UpdateEventLocation | | | | | |

## Dialogs
UpdateEventIntentTrigger
- This trigger should ask the user what they would like to update, then forward to the correct follow-up dialog as necessary

UpdateEventTitleDialog
- This dialog should lookup the specified event using the SearchEvents dialog
- If the 'title' entity was not provided, prompt for the new title
- Call the MS Graph - UpdateEvent action with the new event

UpdateEventDateTimeDialog
- This dialog should lookup the specified event using the SearchEvents dialog
- Use the ResolveDateTime action to extract any datetime entities from the initial utterance
- If no datetime entity is found, prompt for new datetime
- Call the MS Graph - UpdateEvent action with the new event

UpdateEventAddAttendeeDialog
- This dialog should lookup the specified event using the SearchEvents dialog
- The GetContacts dialog should be used to identify the correct contact to add
- The new contact should be added to the existing contacts on the event
- Call the MS Graph - UpdateEvent action with the new event

UpdateEventRemoveAttendeeDialog
- This dialog should lookup the specified event using the SearchEvents dialog
- If the personName and/or emailAddress entities were provided, find the matching contacts in the event.attendees
- If there are no matches, show error message.
- If there are matches, confirm with the user that they want to remove them
- If they say yes, remove them from the event.attendees list
- Call the MS Graph - UpdateEvent action with the new event

UpdateEventLocationDialog
- This dialog should lookup the specified event using the SearchEvents dialog
- If the 'location' entity was not provided, prompt for the new location
- Call the MS Graph - UpdateEvent action with the new event

## Custom Actions
| Name | Input | Output | Notes |
| ------ | ------ | -------- | ------- |
| MS Graph - Get Contacts | | | |
| MS Graph - Search Events | | | |
| MS Graph - Update Event | | | |

## State Properties
| Scope | Property | Type | Notes |
| ------- | --------- | ----- | ------- |
| user | token | TokenResponse object from OAuth Prompt | |
| user | settings.mailboxSettings.timeZone | Time zone string (i.e. Pacific Standard Time) from user's MS Graph MailBoxSettings | |
| conversation | event | The event item should be stored in conversation state for follow-up dialogs | |
| dialog | datetime | DateTime object | |

#### Teams
We may go back and add support for a TaskModule that allows the user to update an event using a form in Teams but that is out of scope for P0.

#### Speech
There is a variation in the flow for the speech channel to confirm the title of the event before continuing. This should only happen in DL Speech.

## Acceptance Criteria
- [ ] Flow should work in both Direct mode (connecting to the skill bot directly) and in Skill mode (connecting through a VA).
- [ ] Should be deployed in a testing environment for review.
- [ ] Flow should work and render correctly in Teams on Desktop, Web, Mobile, and Mac. There may be additional work on cards and other attachments to ensure compatibility.
- [ ] Flow should work and render correctly in Web Chat.
- [ ] Flow should work in Direct Line Speech and should follow the design for speech scenarios.
- [ ] Each response (excluding card responses) should have at least 3 variant LG responses.
- [ ] Each response and card should include speak property provided by design team.
- [ ] Help and Cancel interruptions should be implemented and customized for this dialog where relevant
- [ ] Should include supporting unit tests for both custom actions and declarative dialogs.
- [ ] Language model should be tested against other intents to ensure that there is adequate training data to prevent conflicts
- [ ] Supported languages: English
